### PR TITLE
Fix role permission migration

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
I implemented/fixed _ to achieve _.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/zen_stores/migrations/versions/5994f9ad0489_introduce_role_permissions.py [hunk 1]
```diff
@@ -50,24 +50,24 @@ def upgrade() -> None:
     conn.execute(
         sa.text(
             """
-            DELETE FROM userroleassignmentschema
-            WHERE role_id
-            IN (SELECT id FROM roleschema WHERE name=='admin' OR name=='guest')
+            DELETE U FROM userroleassignmentschema U
+            LEFT JOIN roleschema R ON U.role_id = R.id
```
Comment (fa9r) [inline src/zenml/zen_stores/migrations/versions/5994f9ad0489_introduce_role_permissions.py:54]: Why do you need a left join here? `role_id` should never be `null`, and even if it were, those entries would be filtered by the WHERE clause later, so this seems unnecessary.

```suggestion
            JOIN roleschema R ON U.role_id = R.id
```

File: src/zenml/zen_stores/migrations/versions/5994f9ad0489_introduce_role_permissions.py [hunk 2]
```diff
@@ -50,24 +50,24 @@ def upgrade() -> None:
...
         )
     )
     conn.execute(
         sa.text(
             """
-            DELETE FROM teamroleassignmentschema
-            WHERE role_id
-            IN (SELECT id FROM roleschema WHERE name=='admin' OR name=='guest')
+            DELETE T FROM teamroleassignmentschema T
+            LEFT JOIN roleschema R ON T.role_id = R.id
```
Comment (fa9r) [inline src/zenml/zen_stores/migrations/versions/5994f9ad0489_introduce_role_permissions.py:63]: Same as above, LEFT JOIN seems unnecessary.

```suggestion
            JOIN roleschema R ON T.role_id = R.id
```

**Review by fa9r — APPROVED** [review]
Very happy this is finally fixed so I can merge develop into my branches again 😄 

I am curious though about why the migration failed before. `WHERE x = a OR x = b` should be valid syntax in all SQL versions, no? Perhaps it's because of the `==`?

EDIT: Actually I retract my approval here, if the issue is indeed the `==` instead of `=` then I would prefer the old implementation

**Review by fa9r — COMMENTED** [review]
Actually I retract my approval here, if the issue is indeed the `==` instead of `=` then I would prefer the old implementation

**Review by fa9r — CHANGES_REQUESTED** [review]
See comment above

**Review by fa9r — APPROVED** [review]
LGTM

**Review by stefannica — APPROVED** [review]
Why did we need an `__init__.py` in the `versions` folder ? that does not need to be a python package AFAICT.

**Pull request merged**